### PR TITLE
Migrate default bucket options to `default-options` composite property

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

**Review by dimas-b — APPROVED** [review]

**Comment by snazy** [comment]
Moving to "draft" for now, because I'm still fighting against Helm and update the docs according to this change, which is a bit more than I expected.

**Comment by snazy** [comment]
Meh - once I write about the fight against Helm, CI passes... karma I guess

**Review by dimas-b — COMMENTED** [review]
The PR LGTM :+1: but I'll defer to @adutra for Helm review :)

File: helm/nessie/templates/_helpers.tpl [hunk 1]
```diff
@@ -101,7 +101,7 @@ Apply Nessie Catalog (Iceberg REST) options.
 {{- $map := index . 1 -}}{{/* the destination map */}}
 {{- with $root -}}
 {{- $_ := set $map "nessie.catalog.default-warehouse" .defaultWarehouse -}}
-{{- $_ := set $map "nessie.catalog.object-stores.health-check.enabled" .objectStoresHealthCheckEnabled -}}
+{{- $__ := set $map "nessie.catalog.object-stores.health-check.enabled" .objectStoresHealthCheckEnabled -}}
```
Comment (adutra) [inline helm/nessie/templates/_helpers.tpl:104]:
```suggestion
{{- $_ = set $map "nessie.catalog.object-stores.health-check.enabled" .objectStoresHealthCheckEnabled -}}
```

File: helm/nessie/values.yaml [hunk 2]
```diff
@@ -267,60 +276,67 @@ catalog:
...
+      # -- Per-bucket GCS settings. Override the general settings above.
+      buckets: []
+      # - name: bucket1
+      #   authType: ACCESS_TOKEN
+      #   oauth2TokenSecret:
+      #     - name: gcs-creds
+      #       key: token
+      #       expiresAt: expiresAt
 
       # -- GCS transport settings.
```
Comment (adutra) [inline helm/nessie/values.yaml:341]:
```suggestion
      # -- GCS transport settings. Not overridable on a per-bucket basis.
```

File: helm/nessie/values.yaml [hunk 3]
```diff
@@ -346,49 +362,43 @@ catalog:
...
+      # -- Per-filesystem ADLS settings. Override the general settings above.
+      filesystems: []
+      # - name: filesystem1
+      #   endpoint: http://localhost/adlsgen2/bucket
+      #   accountSecret:
+      #     name: adls-account-secret
+      #     accountName: accountName
+      #     accountKey: accountKeyRef
 
       # -- ADLS transport settings. Only some settings can be overridden on a per-filesystem basis.
```
Comment (adutra) [inline helm/nessie/values.yaml:403]:
```suggestion
      # -- ADLS transport settings. Not overridable on a per-bucket basis.
```

File: helm/nessie/values.yaml [hunk 4]
```diff
@@ -171,79 +171,88 @@ catalog:
...
+          externalId: ~
+          # -- A higher bound estimate of the expected duration of client "sessions" working with data in
+          # this bucket. A session, for example, is the lifetime of an Iceberg REST catalog object on
+          # the client side. This value is used for validating expiration times of credentials
+          # associated with the warehouse. If unset, a default of one hour is assumed.
+          clientSessionDuration: ~
+          # -- The time period to subtract from the S3 session credentials (assumed role credentials)
+          # expiry time to define the time when those credentials become eligible for refreshing.
+          # Not overridable on a per-bucket basis. The default is PT5M (5 minutes).
+          sessionCredentialRefreshGracePeriod: ~  # PT5M
```
Comment (adutra) [inline helm/nessie/values.yaml:229]: These last 3 options are not overridable per bucket. Maybe move them to `transport` or some other section outside `defaultOptions`?

File: helm/nessie/templates/_helpers.tpl (hunk 1 above)
Comment (snazy) [reply helm/nessie/templates/_helpers.tpl:104]: IJ warns about a duplicate var here

File: helm/nessie/values.yaml (hunk 4 above)
Comment (snazy) [reply helm/nessie/values.yaml]: Moved to a new section

File: helm/nessie/templates/_helpers.tpl (hunk 1 above)
Comment (adutra) [reply helm/nessie/templates/_helpers.tpl:104]: I don't have any warning. We are not declaring a new var (`:=`), we are reassigning `$_` (`=`).

Comment (snazy) [reply helm/nessie/templates/_helpers.tpl:104]: image

Comment (adutra) [reply helm/nessie/templates/_helpers.tpl:104]: Please change `:=` to `=`

**Review by adutra — APPROVED** [review]

**Pull request merged**